### PR TITLE
feat: add fake review detection project using BERT and XAI

### LINKED ISSUE #1649: [Project Addition] Fake Review Detection System using BERT & Explainable AI (XAI)
Project Proposal: Fake Review Detection System using BERT & Explainable AI (XAI)

 Project Overview
Online platforms like Amazon, Yelp, and Google are flooded with fake/spam reviews 
that mislead consumers and damage businesses. This project builds an advanced NLP 
pipeline to detect fake reviews using fine-tuned BERT, and uses Explainable AI (LIME/SHAP) 
to interpret WHY the model flagged a review as fake — making it production-relevant.

 Objectives
 Binary classify reviews as **Genuine** or **Fake/Spam**
 Fine-tune a pre-trained **BERT** model on labeled review data
 Apply **LIME / SHAP** to generate word-level explanations for model predictions
 Compare BERT performance against baseline models (TF-IDF + Logistic Regression, LSTM)
 Visualize results with confusion matrix, ROC curve, and SHAP summary plots

 Dataset
**Yelp Fake Review Dataset** (publicly available on Kaggle)
 OR **Amazon Product Reviews** with verified/unverified purchase labels as proxy

 Tech Stack
| Tool | Purpose |
|------|---------|
| Python | Core language |
| HuggingFace Transformers | BERT fine-tuning |
| SHAP / LIME | Explainability |
| scikit-learn | Baseline models & metrics |
| Matplotlib / Seaborn | Visualizations |
| Pandas / NumPy | Data preprocessing |

 Proposed Folder Structure

Fake_Review_Detection/
│
├── fake_review_detection.ipynb   # Main notebook
├── requirements.txt
└── README.md

Deliverables
 [ ] EDA on the review dataset (length distribution, label balance, word clouds)
 [ ] Baseline model: TF-IDF + Logistic Regression
 [ ] Advanced model: Fine-tuned BERT (bert-base-uncased)
 [ ] SHAP / LIME explanation plots for sample predictions
 [ ] Performance comparison table (Accuracy, F1, AUC-ROC)
 [ ] Well-documented README with results and setup instructions

Why This Project?
 Fake review detection is a **real-world, industry-relevant problem**
 Combining BERT + XAI makes this stand out from typical sentiment analysis projects
 Not currently present in the ML-CaPsule repository
 Aligns with the repo's goal of interview-ready, practical ML projects

 Contributor Note
I would like to work on this as part of **GSSoC 2026**. 
Please assign this issue to me if approved. Happy to discuss the approach or 
share a draft notebook before full submission!

### PR DESCRIPTION
**Pull request by sureshsuriya**
## Summary

This PR adds a complete end-to-end NLP project, **Fake Review Detection using BERT and Explainable AI (XAI)**, to the ML-CaPsule repository. The project demonstrates how transformer-based models can be used to identify deceptive online reviews while providing interpretable predictions through Explainable AI techniques.

### 📁 New Project

* Added a new project directory: `Fake_Review_Detection_BERT_XAI/`

### 📓 Jupyter Notebook

Implemented a fully documented notebook covering the complete machine learning workflow:

* Data loading and preprocessing
* Exploratory Data Analysis (EDA)
* Text cleaning and preparation
* Baseline model using **TF-IDF + Logistic Regression**
* Fine-tuning **bert-base-uncased** using Hugging Face Transformers
* Model evaluation using:

  * Accuracy
  * Precision
  * Recall
  * F1-score
  * ROC-AUC
  * Confusion Matrix
  * ROC Curve
* Explainable AI (XAI) using **LIME** and **SHAP** to visualize word-level feature importance and explain model predictions
* Performance comparison between the baseline model and the BERT model

### 📄 Documentation

Added a comprehensive `README.md` including:

* Project overview
* Dataset information
* Installation and setup instructions
* Project workflow
* Model architecture
* Evaluation methodology
* XAI overview
* Expected outputs and visualizations

### 📦 Dependencies

Added a dedicated `requirements.txt` containing all required libraries for data preprocessing, model training, visualization, and explainability.

## Verification

* Verified the notebook structure and organization.
* Verified dataset loading and preprocessing workflow.
* Verified the implementation flow for both the baseline and BERT pipelines.
* Verified the documentation and project structure.
* Confirmed that all changes are contained within the `Fake_Review_Detection_BERT_XAI/` directory.

## Scope

This PR contains **only the changes related to Issue #1649** and does not introduce any unrelated modifications.

Closes #1649

### CODE REVIEW

File: Fake_Review_Detection_BERT_XAI/fake_review_detection.ipynb
Comment (Sibam-Paul) [inline Fake_Review_Detection_BERT_XAI/fake_review_detection.ipynb:1]: run all the cells @sureshsuriya

File: Fake_Review_Detection_BERT_XAI/README.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+- **Labels:** 
+  - `truthful` (800 reviews from TripAdvisor) - mapped to `0`
+  - `deceptive` (800 reviews generated via Mechanical Turk) - mapped to `1`
+- **Balanced Class Distribution:** 50% Genuine, 50% Fake.
+- **Data Columns:**
+  - `deceptive`: Target category (`truthful` / `deceptive`).
+  - `hotel`: The hotel name (20 Chicago hotels represented).
+  - `polarity`: Sentiment direction (`positive` / `negative`).
+  - `source`: Platform source (`TripAdvisor` / `MTurk` / `Web`).
+  - `text`: Raw review text.
```
Comment (Sibam-Paul) [inline Fake_Review_Detection_BERT_XAI/README.md:46]: data the link of the readme @sureshsuriya

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@sureshsuriya make the changes

**Review by Sibam-Paul — APPROVED** [review]
LGTM!

File: Fake_Review_Detection_BERT_XAI/fake_review_detection.ipynb
Comment (Sibam-Paul) [inline Fake_Review_Detection_BERT_XAI/fake_review_detection.ipynb:1]: Ig the plots are not showing check @sureshsuriya please

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@sureshsuriya the plots are not showing

**Comment by sureshsuriya** [comment]
Hi @Sibam-Paul, I've fixed the issue and updated the notebook. The plots are now showing correctly on GitHub. Could you please take another look

File: .github/workflows/shiftleft-analysis.yml
Comment (Sibam-Paul) [inline .github/workflows/shiftleft-analysis.yml:1]: Remove the this file from the commit @sureshsuriya

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@sureshsuriya remove the shiftleft analysis commit we don't need it 
Else ok
